### ISSUE TITLE
[BUG] module "rule" issues with rules that does not return parameter "properties.disabled"

### ISSUE THREAD

**Issue by msekania** [tribe29/ansible-collection-tribe29.checkmk#290]
**Describe the bug**

Some of the rules do not return the "properties.disabled" parameter, but the module checks this and results in an error.

e.g.:
```json
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [],
        "service_description": {
            "match_on": [
                "Check_MK HW/SW Inventory$"
            ],
            "operator": "one_of"
        },
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "Restrict HW/SW-Inventory to once a day"
    },
    "ruleset": "extra_service_conf:check_interval",
    "value_raw": "1440.0"
},
"extensions": {
    "conditions": {
        "host_labels": [
            {
                "key": "cmk/docker_object",
                "operator": "is",
                "value": "container"
            }
        ],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "Make all docker container host states base on the \"Docker container status\" service"
    },
    "ruleset": "host_check_commands",
    "value_raw": "('service', 'Docker container status')"
},
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [
            {
                "key": "criticality",
                "operator": "is_not",
                "value": "offline"
            }
        ],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "Do not monitor hosts with the tag \"offline\""
    },
    "ruleset": "only_hosts",
    "value_raw": "True"
},
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "Perform every two hours a service discovery"
    },
    "ruleset": "periodic_discovery",
    "value_raw": "{'check_interval': 120.0, 'severity_unmonitored': 1, 'severity_vanished': 0, 'severity_new_host_label': 1}"
},
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [
            {
                "key": "snmp",
                "operator": "is",
                "value": "snmp"
            },
            {
                "key": "snmp_ds",
                "operator": "is_not",
                "value": "snmp-v1"
            }
        ],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "Hosts with the tag \"snmp-v1\" must not use bulkwalk"
    },
    "ruleset": "bulkwalk_hosts",
    "value_raw": "True"
},
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {
        "description": "All management boards use SNMPv2 and bulk walk"
    },
    "ruleset": "management_bulkwalk_hosts",
    "value_raw": "True"
},
"extensions": {
    "conditions": {
        "host_labels": [],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {},
    "ruleset": "extra_host_conf:notification_options",
    "value_raw": "'d,r,f,s'"
},
"extensions": {
    "conditions": {
        "host_labels": [
            {
                "key": "cmk/check_mk_server",
                "operator": "is",
                "value": "yes"
            }
        ],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {},
    "ruleset": "inventory_df_rules",
    "value_raw": "{'ignore_fs_types': ['tmpfs', 'nfs', 'smbfs', 'cifs', 'iso9660'], 'never_ignore_mountpoints': ['~.*/omd/sites/[^/]+/tmp$']}"
},
"extensions": {
    "conditions": {
        "host_labels": [
            {
                "key": "cmk/docker_object",
                "operator": "is",
                "value": "node"
            }
        ],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 0,
    "properties": {},
    "ruleset": "active_checks:cmk_inv",
    "value_raw": "{'status_data_inventory': True}"
},
"extensions": {
    "conditions": {
        "host_labels": [
            {
                "key": "cmk/check_mk_server",
                "operator": "is",
                "value": "yes"
            }
        ],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 1,
    "properties": {},
    "ruleset": "active_checks:cmk_inv",
    "value_raw": "{'status_data_inventory': True}"
},
"extensions": {
    "conditions": {
        "host_labels": [
            {
                "key": "cmk/kubernetes",
                "operator": "is",
                "value": "yes"
            }
        ],
        "host_tags": [],
        "service_labels": []
    },
    "folder": "/",
    "folder_index": 2,
    "properties": {},
    "ruleset": "active_checks:cmk_inv",
    "value_raw": "{'status_data_inventory': True}"
},
```

**Component Name**

`rule`

**Ansible Version**

```console
ansible [core 2.14.1]
  config file = None
  python version = 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0] (/usr/bin/python3)
  jinja version = 3.0.3
  libyaml = True
```

**Checkmk Version**

```console
2.1.0p11 (CRE)
```

**Collection Version**

```console
tribe29.checkmk      0.18.0 
```
**Environment**

**To Reproduce**

*example playbook*:
```yaml
- name: Test rule
  hosts: localhost
  gather_facts: false
  vars:
    site: "my_site"
    server_url: "my_server_url"
    automation_user: "my_automation_user"
    automation_secret: "my_automation_user_secret"

    some_rules: [
      "extensions": {
        "conditions": {
          "host_labels": [],
          "host_tags": [],
          "service_description": {
            "match_on": [
              "Check_MK HW/SW Inventory$"
            ],
            "operator": "one_of"
          },
          "service_labels": []
        },
        "folder": "/",
        "folder_index": 0,
        "properties": {
          "description": "Restrict HW/SW-Inventory to once a day"
        },
        "ruleset": "extra_service_conf:check_interval",
        "value_raw": "1440.0"
      },
      "extensions": {
        "conditions": {
          "host_labels": [
            {
              "key": "cmk/docker_object",
              "operator": "is",
              "value": "container"
            }
          ],
          "host_tags": [],
          "service_labels": []
        },
        "folder": "/",
        "folder_index": 0,
        "properties": {
          "description": "Make all docker container host states base on the \"Docker container status\" service"
        },
        "ruleset": "host_check_commands",
        "value_raw": "('service', 'Docker container status')"
      },
    ]

  tasks:
    - name: "Create rest of rules"
      tribe29.checkmk.rule:
        server_url: "{{ server_url }}"
        site: "{{ site }}"
        automation_user: "{{ automation_user }}"
        automation_secret: "{{ automation_secret }}"
        ruleset: "{{ item.extensions.ruleset }}"
        rule: "{{ rule }}"
        state: "present"
      run_once: true
      delegate_to: localhost
      loop: "{{ some_rules }}"
      vars:
        rule:
          folder: "{{ item.extensions.folder }}"
          properties: "{{ item.extensions.properties }}"
          conditions: "{{ item.extensions.conditions }}"
          value_raw: "{{ item.extensions.value_raw }}"
```

**Expected behavior**

**Actual behavior**

fails in normal and check mode also. 

```console
PLAY [Test rule] ********************************************************************************************************************************************************************************************************************************************

TASK [Create rest of rules] *********************************************************************************************************************************************************************************************************************************
Freitag 24 März 2023  10:17:52 +0100 (0:00:00.028)       0:00:00.028 ********** 
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'disabled'
failed: [localhost] (item={'extensions': {'conditions': {'host_labels': [], 'host_tags': [], 'service_description': {'match_on': ['Check_MK HW/SW Inventory$'], 'operator': 'one_of'}, 'service_labels': []}, 'folder': '/', 'folder_index': 0, 'properties': {'description': 'Restrict HW/SW-Inventory to once a day'}, 'ruleset': 'extra_service_conf:check_interval', 'value_raw': '1440.0'}}) => {"ansible_loop_var": "item", "changed": false, "item": {"extensions": {"conditions": {"host_labels": [], "host_tags": [], "service_description": {"match_on": ["Check_MK HW/SW Inventory$"], "operator": "one_of"}, "service_labels": []}, "folder": "/", "folder_index": 0, "properties": {"description": "Restrict HW/SW-Inventory to once a day"}, "ruleset": "extra_service_conf:check_interval", "value_raw": "1440.0"}}, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 107, in <module>\n  File \"<stdin>\", line 99, in _ansiballz_main\n  File \"<stdin>\", line 47, in invoke_module\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_rj2db5h1/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 528, in <module>\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_rj2db5h1/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 524, in main\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_rj2db5h1/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 511, in run_module\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_rj2db5h1/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 300, in create_rule\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_rj2db5h1/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 286, in get_existing_rule\nKeyError: 'disabled'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: KeyError: 'disabled'
failed: [localhost] (item={'extensions': {'conditions': {'host_labels': [{'key': 'cmk/docker_object', 'operator': 'is', 'value': 'container'}], 'host_tags': [], 'service_labels': []}, 'folder': '/', 'folder_index': 0, 'properties': {'description': 'Make all docker container host states base on the "Docker container status" service'}, 'ruleset': 'host_check_commands', 'value_raw': "('service', 'Docker container status')"}}) => {"ansible_loop_var": "item", "changed": false, "item": {"extensions": {"conditions": {"host_labels": [{"key": "cmk/docker_object", "operator": "is", "value": "container"}], "host_tags": [], "service_labels": []}, "folder": "/", "folder_index": 0, "properties": {"description": "Make all docker container host states base on the \"Docker container status\" service"}, "ruleset": "host_check_commands", "value_raw": "('service', 'Docker container status')"}}, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 107, in <module>\n  File \"<stdin>\", line 99, in _ansiballz_main\n  File \"<stdin>\", line 47, in invoke_module\n  File \"/usr/lib/python3.10/runpy.py\", line 224, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib/python3.10/runpy.py\", line 96, in _run_module_code\n    _run_code(code, mod_globals, init_globals,\n  File \"/usr/lib/python3.10/runpy.py\", line 86, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_3g37ddpf/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 528, in <module>\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_3g37ddpf/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 524, in main\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_3g37ddpf/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 511, in run_module\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_3g37ddpf/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 300, in create_rule\n  File \"/tmp/ansible_tribe29.checkmk.rule_payload_3g37ddpf/ansible_tribe29.checkmk.rule_payload.zip/ansible_collections/tribe29/checkmk/plugins/modules/rule.py\", line 286, in get_existing_rule\nKeyError: 'disabled'\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
```

**Screenshots**

**Additional context**

**Comment by msekania** [comment]
@geof77, @lgetwan,

We should adjust check conditions in `get_existing_rule` should be adjusted.

If I'll have some elegant solution I'll post it here.

**Comment by robin-tribe29** [comment]
@msekania thanks for your report!
Can you provide a few more details on the error message and when/how this happens exactly?
Also, could you test this against a current patch release of Checkmk?

**Comment by msekania** [comment]
@robin-tribe29 

> Can you provide a few more details on the error message and when/how this happens exactly?
Yes, I have update description

> Also, could you test this against a current patch release of Checkmk?
At the moment unfortunately not.

**Comment by msekania** [comment]
@robin-tribe29, @lgetwan, @geof77 

Here is a quick fix, but let's see carefully if there are some similar issues
 
```python
def get_existing_rule(module, base_url, headers, ruleset, rule):
    # Get rules in ruleset
    rules = get_rules_in_ruleset(module, base_url, headers, ruleset)

    (value_mod, exc) = safe_eval(rule["value_raw"], include_exceptions=True)
    if exc is not None:
        exit_failed(module, "value_raw in rule has invalid format")

    if rules is not None:
        # Loop through all rules
        for r in rules.get("value"):
            (value_api, exc) = safe_eval(
                r["extensions"]["value_raw"], include_exceptions=True
            )
            if exc is not None:
                exit_failed(module, "Error deserializing value_raw from API")
            if (
                r["extensions"]["folder"] == rule["folder"]
                and r["extensions"]["conditions"] == rule["conditions"]
                # and r["extensions"]["properties"]["disabled"]
                # == rule["properties"]["disabled"]
                and r["extensions"]["properties"].get("disabled", "")
                == rule["properties"].get("disabled", "")
                and value_api == value_mod
            ):
                # If they are the same, return the ID
                return r

    return None

```

**Comment by lgetwan** [comment]
@msekania: At first sight, your suggested change fixes the problem and doesn't cause other problems.
@geof77: Do you have any concerns about the suggested change?

**Comment by robin-tribe29** [comment]
Fixed in #296.
